src/appshell/internal/startupscenario.cpp

```diff
@@ -181,8 +303,17 @@ void StartupScenario::onStartupPageOpened(StartupModeType modeType)
     } break;
     }
 
+    const Version welcomeDialogLastShownVersion(configuration()->welcomeDialogLastShownVersion());
+    const Version currentMuseScoreVersion(configuration()->museScoreVersion());
+    if (true) {
+        configuration()->setWelcomeDialogShowOnStartup(true); // override user preference
+        configuration()->setWelcomeDialogLastShownIndex(-1); // reset
+    }
+
     if (!configuration()->hasCompletedFirstLaunchSetup()) {
         interactive()->open(FIRST_LAUNCH_SETUP_URI);
+    } else if (configuration()->welcomeDialogShowOnStartup() && !museSoundsUpdateScenario()->hasUpdate()) {
+        showWelcomeDialog();
```

Comment:
Could you please explain at what point are the program update and musesounds update dialogs will be shown?